provider/identityprovider/identity_provider_resource.go

```diff
@@ -581,7 +583,76 @@ func (r *IdentityProviderResource) Read(ctx context.Context, request resource.Re
 
 func (r *IdentityProviderResource) Update(ctx context.Context, request resource.UpdateRequest,
 	response *resource.UpdateResponse) {
-	response.Diagnostics.AddError("IDP Update not supported.", "This RHCS provider version does not support updating an existing IDP")
+	// Get the state:
+	state := &IdentityProviderState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	if state.HTPasswd == nil {
+		response.Diagnostics.AddError("IDP Update not supported for non-HTPasswd IDPs.",
+			"This RHCS provider version does not support updating an existing IDP")
+		return
+	}
+
+	// Get the plan:
+	plan := &IdentityProviderState{}
+	diags = request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	plan.ID = state.ID
+
+	resource := r.collection.Cluster(state.Cluster.ValueString()).IdentityProviders().
+		IdentityProvider(state.ID.ValueString())
+
+	if !reflect.DeepEqual(state.HTPasswd.Users, plan.HTPasswd.Users) {
+		stateUserMap, err := HTPasswdUserListToStringMaps(ctx, state.HTPasswd.Users, resource)
+		if err != nil {
+			response.Diagnostics.AddError(
+				"Can't get identity provider user ID",
+				fmt.Sprintf(
+					"Can't get identity provider user ID with identifier for "+
+						"cluster '%s': %v",
+					state.Cluster.ValueString(), err,
+				),
+			)
+			return
+		}
+		planUserMap, err := HTPasswdUserListToStringMaps(ctx, plan.HTPasswd.Users, resource)
+		if err != nil {
+			response.Diagnostics.AddError(
+				"Can't get identity provider user ID",
+				fmt.Sprintf(
+					"Can't get identity provider user ID with identifier for "+
+						"cluster '%s': %v",
+					state.Cluster.ValueString(), err,
+				),
+			)
+			return
+		}
+
+		patchParams := htpasswd.PatchParams{
+			ctx, stateUserMap, planUserMap, resource,
+			[]string{}, state.Cluster.ValueString(), response,
+		}
+
+		var ok bool
+		patchParams.RemovedUsers, ok = htpasswd.DeleteUserFromState(patchParams)
+		if !ok {
+			return
+		}
+
+		ok = htpasswd.PatchOrAddUserInState(patchParams)
+		if !ok {
+			return
+		}
+	}
```

Comment:
What do you say about moving all this code to htpasswd.go as well?
This code is relted only to htpasswd
Call to function like:
```
updateHTPasswd(state.HTPasswd, plan.HTPasswd)
```

Reply:
SGTM